_includes/macros.html

```diff
@@ -166,11 +166,11 @@ <h1 class="h5 {{'u-visually-hidden' if hide_heading}}">{{ text }}</h1>
    ========================================================================== #}
 
 {% macro format_address(contact) %}
-    {{ contact.street if contact.street }}<br>
-    {{ contact.street_2 + '<br>'|safe if contact.street_2 }}
-    {{ contact.city + ',' if contact.city }}
-    {{ (contact.state + '&nbsp;')|safe if contact.state }}
-    {{ contact.zip_code if contact.zip_code }}
+    {{ contact.street if contact.street else '' }}<br>
+    {{ contact.street_2 + '<br>'|safe if contact.street_2 else '' }}
+    {{ contact.city + ',' if contact.city else '' }}
+    {{ (contact.state + '&nbsp;')|safe if contact.state else '' }}
+    {{ contact.zip_code if contact.zip_code else '' }}
```

Comment:
OK here's what's happening: By default, if Django is in debug mode, it will instantiate the Jinja2 environment using the DebugUndefined object-- which in a block like the one above, is pretty noisy:

![screen shot 2015-06-23 at 11 01 59 am](https://cloud.githubusercontent.com/assets/235397/8309230/53aa7c3e-1997-11e5-90d5-979f8b474219.png)

(and that's after removing just ones of the else's) 

If debug is false, those warnings go away.

There are ways to go about customizing this (say, log those kind of messages to the console instead of the browser), but in general, this feels like the system working.

Django's code:
https://github.com/django/django/blob/master/django/template/backends/jinja2.py#L32

Undefined docs:
http://jinja.pocoo.org/docs/dev/api/#jinja2.Undefined

DebugUndefined docs:
http://jinja.pocoo.org/docs/dev/api/#jinja2.DebugUndefined